mlock.c calls into a function from libgloss without including the header.
We add the relative path to the header here instead of adding the libgloss
includes to include paths to keep the patch short.
--- a/newlib/libc/sys/or1k/mlock.c	2023-12-31 18:00:18.000000000 +0100
+++ b/newlib/libc/sys/or1k/mlock.c	2024-05-09 13:26:08.523124156 +0200
@@ -17,6 +17,7 @@
 
 #include <reent.h>
 #include <stdint.h>
+#include "../../../../libgloss/or1k/include/or1k-support.h"
 
 /* Lock calls from different cores, but allows recursive calls from the same
  * core. The lock is not only atomic to other cores calling malloc, but also
